### PR TITLE
[Easy] misc clippy fixes

### PR DESCRIPTION
**Pull request by fleupold**
This PR fixes a bunch of clippy suggestions which are not land blocking (since they are minor). More specifically:

- trait objects without an explicit `dyn` are deprecated
- &vec![...] == &[...]
- complex types in mocks should be extracted (typedef)
- new methods without arguments should be implemented via Default trait
- don't use clone on copy types

### CODE REVIEW

File: driver/src/order_driver.rs [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
 
 use dfusion_core::database::DbInterface;
 use dfusion_core::models::{
-    AccountState,
-    ConcatenatingHashable,
-    Order,
-    RollingHashable,
-    Serializable,
-    Solution,
+    AccountState, ConcatenatingHashable, Order, RollingHashable, Serializable, Solution,
```
Comment (bh2smith) [inline driver/src/order_driver.rs]: I kinda prefer the look of the vertical/stacked imports here...

File: driver/src/order_driver.rs [hunk 2]
```diff
@@ -85,8 +80,8 @@ impl<'a, D: DbInterface, C: SnappContract> OrderProcessor<'a, D, C> {
                     }
                     return Err(DriverError::new(
                         &format!("Cannot find saved bid for auction slot {:?}", slot),
-                        ErrorKind::StateError
-                    ))
+                        ErrorKind::StateError,
+                    ));
```
Comment (bh2smith) [inline driver/src/order_driver.rs]: How is it even possible that we did not have a semi-colon here before?

File: driver/src/price_finding/linear_optimization_price_finder.rs [hunk 3]
```diff
@@ -333,9 +331,9 @@ pub mod tests {
 
         let expected_solution = models::Solution {
             surplus: U256::from_dec_str("15854632034944469292777429010439194350").ok(),
-            prices: vec![14024052566155238000, 1526784674855762300],
-            executed_sell_amounts: vec![0, 318390084925498118944],
-            executed_buy_amounts: vec![0, 95042777139162480000],
+            prices: vec![14_024_052_566_155_238_000, 1_526_784_674_855_762_300],
```
Comment (bh2smith) [inline driver/src/price_finding/linear_optimization_price_finder.rs]: Nice one!

File: driver/src/withdraw_driver.rs [hunk 4]
```diff
@@ -15,10 +15,9 @@ where
     let withdraw_slot = contract.get_current_withdraw_slot()?;
 
     info!("Current top withdraw_slot is {:?}", withdraw_slot);
-    let slot = find_first_unapplied_slot(
-        withdraw_slot,
-        &|i| contract.has_withdraw_slot_been_applied(i),
-    )?;
+    let slot = find_first_unapplied_slot(withdraw_slot, &|i| {
```
Comment (bh2smith) [inline driver/src/withdraw_driver.rs]: I guess this looks better...

File: driver/src/contract.rs [hunk 5]
```diff
@@ -322,7 +322,10 @@ impl SnappContract for SnappContractImpl {
         new_state: H256,
         prices_and_volumes: Vec<u8>,
     ) -> Result<()> {
-        debug!("Applying Auction with result bytes: {:?}", &prices_and_volumes);
```
Comment (bh2smith) [inline driver/src/contract.rs]: I don't see that this was exceeding the line width (considering there are others added here that were longer). This seems like an unnecessary change.

File: driver/src/order_driver.rs (hunk 2 above)
Comment (fleupold) [reply driver/src/order_driver.rs]: I guess for return statements, you don't need to have a `;`

File: driver/src/contract.rs (hunk 5 above)
Comment (fleupold) [reply driver/src/contract.rs]: rustfmt breaks it for me whenever I save
